src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -2311,6 +2311,73 @@ public async Task responses_to_named_requests_can_be_accessed_as_symbols_in_late
         secondResult.Events.Should().NotContainErrors();
     }
 
+    [Fact]
+    public async Task Response_headers_for_named_requests_can_be_accessed_correctly()
+    {
+
+        var responseHandler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            response.RequestMessage = message;
+            var contentString = @"
+             {
+    ""headers"": {
+        ""Accept"": ""text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"",
+        ""Accept-Encoding"": ""gzip, deflate, br, zstd"",
+        ""Accept-Language"": ""en-US,en;q=0.9"",
+        ""Host"": ""httpbin.org"",
+        ""Priority"": ""u=0, i"",
+        ""Sec-Ch-Ua"": ""\""Chromium\"";v=\""128\"", \""Not;A=Brand\"";v=\""24\"", \""Microsoft Edge\"";v=\""128\"""",
+        ""Sec-Ch-Ua-Mobile"": ""?0"",
+        ""Sec-Ch-Ua-Platform"": ""\""Windows\"""",
+        ""Sec-Fetch-Dest"": ""document"",
+        ""Sec-Fetch-Mode"": ""navigate"",
+        ""Sec-Fetch-Site"": ""none"",
+        ""Sec-Fetch-User"": ""?1"",
+        ""Upgrade-Insecure-Requests"": ""1"",
+        ""User-Agent"": ""Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 Edg/128.0.0.0"",
+        ""X-Amzn-Trace-Id"": ""Root=1-66e9f16e-7afea6f643e754f854c57d85""
+    }
+}
+            ";
+            response.Content = new StringContent(contentString, Encoding.UTF8, "application/json");
+            response.Headers.Add("server", "gunicorn/19.9.0");
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(responseHandler);
+        // var client = new HttpClient();
+        using var kernel = new HttpKernel("http", client);
+
+        var code = """
+            @baseUrl = https://httpbin.org/headers
+
+            # @name binHeader
+            GET {{baseUrl}}
+
+            ###
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+
+        var secondCode = $$$"""
+            GET https://httpbin.org/headers
+            Detectedserver: {{binHeader.response.headers.Server}}
```

Comment:
Looks like this expression validates the case where path depth is 4

Do we already have other tests that cover cases like below -
* Path depth of 5
* Negative tests for path depth of 6 or more
* Negative test where someone tries to use json path / xpath after `binheader.response.headers` or after `binheader.request.headers.someHeaderName`
* Negative test for path depth of 3 or less. For example,
  * `{{binheader.response.headers}}`
  * `{{binheader.request}}`
  * `{{binheader}}`
* Negative tests for cases where the first, second and third parts contain garbage. For example,
  * 1st part is not a valid named request
  * 1st part maps to a named request, but 2nd part is neither `request` nor `response`, or 
  * 1st and 2nd parts are valid, but the 3rd part is neither `headers` not `body` and so on...

Reply:
I had some ideas for a different design for this piece that might better accommodate some of these complexities. It's probably worth finding some time to discuss.

Reply:
I'll work on a test for the depth of 5, a negative test for the depth of 6, and a test where the depth is correct but the header doesn't exist but all of the other cases are handled by other tests already